src/Microsoft.Azure.WebJobs.Host/Blobs/Triggers/BlobTriggerAttributeBindingProvider.cs

```diff
@@ -160,6 +160,11 @@ private static IBlobArgumentBindingProvider CreateProvider(IEnumerable<Type> clo
             IStorageAccount hostAccount = await _accountProvider.GetStorageAccountAsync(context.CancellationToken);
             IStorageAccount dataAccount = await _accountProvider.GetStorageAccountAsync(context.Parameter, context.CancellationToken, _nameResolver);
 
+            if (dataAccount.Type == StorageAccountType.Premium)
```

Comment:
It makes more sense for this logic to be applied at the ScriptHost level. At the webjobs level there is no problem with the trigger.